corehq/ex-submodules/pillowtop/processors/form.py

```diff
@@ -104,4 +118,14 @@ def mark_latest_submission(domain, user_id, app_id, build_id, version, metadata,
         last_submission.build_version = app_version_info.build_version
         last_submission.commcare_version = app_version_info.commcare_version
 
+        if app_version_info.build_version:
+            update_latest_builds(user, app_id, received_on_datetime, app_version_info.build_version)
+
+        if _last_submission_needs_update(user.reporting_metadata.last_submission_for_user,
+                                         received_on_datetime,
+                                         app_version_info.build_version,
+                                         app_version_info.commcare_version):
```

Comment:
Also thinking that the first `if` should be checking against the global submission and not the app specific one.